controller/backup_target_controller.go

```diff
@@ -213,8 +213,26 @@ func getLoggerForBackupTarget(logger logrus.FieldLogger, backupTarget *longhorn.
 	)
 }
 
+func getAvailableDataEngine(ds *datastore.DataStore) (longhorn.DataEngineType, error) {
+	dataEngines := ds.GetDataEngines()
+	if len(dataEngines) > 0 {
+		for _, dataEngine := range []longhorn.DataEngineType{longhorn.DataEngineTypeV2, longhorn.DataEngineTypeV1} {
+			if _, ok := dataEngines[dataEngine]; ok {
+				return dataEngine, nil
+			}
+		}
+	}
+
+	return "", errors.New("no data engine available")
+}
```

Comment:
note: just need to choose any available one, as it's just for operating the remote backup store.